CHANGELOG/CHANGELOG-3.4.md

```diff
@@ -4,7 +4,7 @@ Previous change logs can be found at [CHANGELOG-3.3](https://github.com/etcd-io/
 
 <hr>
 
-## v3.4.37 (TBC)
+## v3.4.37 (2025-04-15)
```

Comment:
Nit: Normally when we add release date we often add the next header section for the following release i.e. `v3.4.38`.  If you have a moment please feel free to add it.

Reply:
Updated!